### PR TITLE
feat(train): implement HTTP endpoint-driven training service and react frontend dashboard console

### PR DESCRIPTION
**Pull request by milesgray**
## Summary
Transforms the training service from a CLI-driven script into an HTTP-enabled training service, and connects it directly to a React frontend dashboard.

## Changes
- **FastAPI Training Service**: Launched asynchronous model training using Pydantic validation and thread workers.
- **REST Endpoints**: Implemented routes for training (`POST /train`), task status check (`GET /tasks/{task_id}`), checkpoints listing (`GET /models`), weights retrieval (`GET /models/{model_id}/download`), and dynamic prediction inference (`POST /models/{model_id}/predict`).
- **Vite React Frontend**: Configured proxying rules, implemented Axios helpers, and updated the `ModelTrainingConsole` to support launching training runs, monitoring progress via live ECharts loss curves, and testing checkpoints in an interactive playground.
- **Bug Fixes**: Corrected an UnboundLocalError in `forecast.py` and imported dotdict correctly in training pipeline tests.

## Testing
- [x] Tests pass locally
- [x] Docs updated

### CODE REVIEW

File: services/train/main.py [hunk 1]
```diff
@@ -1,14 +1,343 @@
...
+                    with open(os.path.join(root, "config.json"), "r") as f:
+                        model_info["config"] = json.load(f)
+                except:
+                    pass
+            models.append(model_info)
+    return models
+
+
+@app.get("/models/{model_id:path}/download")
+def download_model(model_id: str, checkpoints_dir: str = "./checkpoints/"):
```
Comment (milesgray) [inline services/train/main.py:281]: No path resolution or boundary validation is performed on the user-supplied `model_id` path parameter or `checkpoints_dir` query parameter. A malicious request containing `../` sequences or pointing `checkpoints_dir` to system roots could download arbitrary files from the server. Consider resolving the absolute paths using `os.path.abspath()` and asserting that the target path starts with the checkpoints directory path.

File: services/train/main.py [hunk 2]
```diff
@@ -1,14 +1,343 @@
...
+        # Try finding directly
+        if os.path.exists(os.path.join(model_id, "checkpoint.pth")):
+            checkpoint_file = os.path.join(model_id, "checkpoint.pth")
+        else:
+            raise HTTPException(status_code=404, detail=f"Model checkpoint not found for {model_id}")
+    return FileResponse(checkpoint_file, media_type="application/octet-stream", filename=f"{os.path.basename(model_id)}_checkpoint.pth")
+
+
+@app.post("/models/{model_id:path}/predict")
+def run_model_inference(model_id: str, request: PredictRequest, checkpoints_dir: str = "./checkpoints/"):
```
Comment (milesgray) [inline services/train/main.py:290]: Similar to the download endpoint, `run_model_inference` allows directory traversal via `model_id` or `checkpoints_dir`. Validate paths to prevent reading configurations or executing inference using checkpoints outside the checkpoints directory.

File: services/train/main.py [hunk 3]
```diff
@@ -1,14 +1,343 @@
...
+        configs = dotdict(model_config)
+        configs.use_gpu = False
+        model = get_model(configs).float()
+        model.load_state_dict(torch.load(checkpoint_path, map_location="cpu"))
+        model.eval()
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=f"Failed to load model/weights: {str(e)}")
+
+    try:
+        x_tensor = torch.FloatTensor(request.x)
```
Comment (milesgray) [inline services/train/main.py:326]: Verify the shape of `request.x` against `configs.seq_len` and `configs.enc_in` before converting it to a PyTorch tensor, to provide a clear HTTP 400 Bad Request error if input shape is invalid.

**Review by milesgray — COMMENTED** [review]
### Review Summary

#### ⚠️ Medium
- **Directory Traversal vulnerability in model routes**: In `services/train/main.py`, the endpoints `GET /models/{model_id}/download` and `POST /models/{model_id}/predict` join `checkpoints_dir` and the user-controlled `model_id` path parameter without validating that the resulting path stays within the intended checkpoints directory structure. Additionally, `checkpoints_dir` is accepted as an arbitrary query parameter, allowing callers to point to other directories on the filesystem.

#### ℹ️ Low
- **Inference Shape validation**: In `run_model_inference` (`main.py` line 313), `torch.FloatTensor(request.x)` is constructed without validating that the input dimensions match the sequence length (`seq_len`) and input channels (`enc_in`) defined in the model configuration. Though caught by standard exceptions, an explicit shape assertion would return a cleaner HTTP 400 client error.
